rules/manager.go

```diff
@@ -314,14 +325,36 @@ func (g *Group) run(ctx context.Context) {
 	tick := time.NewTicker(g.interval)
 	defer tick.Stop()
 
+	makeStale := func(s bool) {
+		if !s {
+			return
+		}
+		go func(now time.Time) {
+			// Wait for 2 intervals to give the opportunity to renamed rules
+			// to insert new series in the tsdb. At this point if there is a
+			// renamed rule, it should already be started.
+			select {
+			case <-g.managerDone:
+			case <-time.After(2 * g.interval):
+				for _, rule := range g.seriesInPreviousEval {
+					for _, s := range rule {
```

Comment:
You're shadowing `s` here. Why not move the top if out of this function?

Reply:
Done, I have also set g.seriesInPreviousEval to nil so it can be garbage collected before 2 * g.interval